__tests__/components/AddressBarInput.spec.js

```diff
@@ -67,4 +64,112 @@ describe( 'AddressBarInput', () =>
             expect( instance.constructor.name ).toMatch( 'Input' );
         } );
     } );
+
+    describe( 'events', () =>
+    {
+        beforeEach( () =>
+        {
+            store = mockStore( props );
+
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } />
+                </Provider > );
+            instance = wrapper.instance();
+        } );
+
+        afterEach( () =>
+        {
+            wrapper.unmount();
+        } );
+
+        it( 'check on onBlur,handleBlur is called', () =>
+        {
+            const handleBlur = jest.fn();
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } onBlur={ handleBlur } />
+                </Provider > );
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'blur' );
+            expect( handleBlur ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onBlur,onBlur() is called', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'blur' );
+            expect( props.onBlur ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onFocus,handleFocus is called', () =>
+        {
+            const handleFocus = jest.fn();
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } onFocus={ handleFocus } />
+                </Provider > );
+            instance = wrapper.instance();
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'focus' );
+            expect( handleFocus ).toHaveBeenCalled();
+        } );
+
+        it( 'check on onFocus,onFocus() is called', () =>
+        {   
+            wrapper = mount(
+                <Provider store={ store } >
+                    <AddressBarInput { ...props } />
+                </Provider > );
+            instance = wrapper.instance();
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'focus' );
+            expect( props.onFocus ).toHaveBeenCalled();
+        } );
+
+        it( 'check if onKeyPress triggers', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            expect( input.simulate( 'keyPress', { key: 'Enter', keyCode: 13, which: 13 } ) ).toBeTruthy();
+        } );
+
+        it( 'check on onKeyPress if activeTab is called', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.simulate( 'keyPress', { key: 'Enter', keyCode: 13, which: 13 } );
+            expect( props.updateActiveTab ).toHaveBeenCalled();
+        } );
+
+        it( 'check if onChange triggers', () =>
+        {
+            const input = wrapper.find( 'Input' );
+            input.value = '123456';
+            input.simulate( 'change' );
+            expect( input.simulate( 'change' ) ).toBeTruthy();
```

Comment:
what's going on here with simulating change twice? 

I don't see what you're checking.

Reply:
is checking the simulation necessary? that's not our code. The simulation is enzyme code.